extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-preview.js

```diff
@@ -0,0 +1,654 @@
+        this.updateDevicesInHash_();
+      }
+    });
+
+    chipListContainer.appendChild(this.buildAddDeviceButton_());
+    chipListContainer.appendChild(this.buildHelpButton_());
+
+    parseDevices(
+      this.element.getAttribute('devices') || DEFAULT_DEVICES
+    ).forEach((device) => {
+      this.addDevice_(device);
+    });
+  }
+
+  /**
+   * Builds the add device button
+   * @private
+   * @return {!Element}
+   */
+  buildAddDeviceButton_() {
+    const addDeviceButton = buildDeviceChipTemplate(this.element);
+    addDeviceButton.classList.add('i-amphtml-story-dev-tools-add-device');
+    addDeviceButton.classList.add('i-amphtml-story-dev-tools-button');
+    addDeviceButton.querySelector('span').textContent = 'ADD DEVICE';
+    addDeviceButton.addEventListener('click', () => {
+      this.showAddDevicePopup_();
+    });
+    return addDeviceButton;
+  }
+
+  /**
+   * Builds the add device button
+   * @private
+   * @return {!Element}
+   */
+  buildHelpButton_() {
+    const addHelpButton = buildHelpButtonTemplate(this.element);
+    addHelpButton.addEventListener('click', () => {
+      this.showHelpPopup_();
+    });
+    return addHelpButton;
+  }
+
+  /**
+   * Creates a device layout for preview
+   * @private
+   * @param {!DeviceInfo} device
+   * @return {!Element}
+   */
+  buildDeviceLayout_(device) {
+    const deviceLayout = buildDeviceTemplate(this.element);
+    const devicePlayer = deviceLayout.querySelector('amp-story-player');
+    devicePlayer.setAttribute('width', device.width);
+    devicePlayer.setAttribute('height', device.height);
+    const storyA = devicePlayer.querySelector('a');
+    storyA.textContent = 'Story 1';
+    storyA.href = this.storyUrl_;
+    setStyles(devicePlayer, {
+      width: device.width + 'px',
+      height: device.height + 'px',
+    });
+    const playerImpl = new AmpStoryPlayer(this.win, devicePlayer);
+    playerImpl.load();
+    setStyles(
+      deviceLayout.querySelector('.i-amphtml-story-dev-tools-device-screen'),
+      {
+        height: device.deviceHeight
+          ? device.deviceHeight + 'px'
+          : 'fit-content',
+      }
+    );
+    device.player = devicePlayer;
+    return deviceLayout;
+  }
+
+  /**
+   * @private
+   * @param {DeviceInfo} deviceSpecs
+   */
+  addDevice_(deviceSpecs) {
+    const deviceLayout = this.buildDeviceLayout_(deviceSpecs, this.storyUrl_);
+    deviceSpecs.element = deviceLayout;
+    this.element.appendChild(deviceLayout);
+    deviceSpecs.chip = this.addDeviceChip_(deviceSpecs);
+    this.devices_.push(deviceSpecs);
+    this.onLayoutMeasure();
+    setStyles(deviceLayout, {'opacity': '1'});
+    setTimeout(() => {
+      setStyles(deviceLayout, {
+        'transition': 'transform ease-out 0.15s',
+      });
+    }, 150);
+  }
+
+  /**
+   * @private
+   * @param {DeviceInfo} device
+   */
+  removeDevice_(device) {
+    device.chip.setAttribute('inactive', '');
+    setTimeout(() => {
+      device.chip.remove();
+    }, 200);
```

Comment:
why 200? how do we make sure we never break this if we change something else?

Reply:
It's the `transition` in the chip, that when you set to [inactive] in the bottom list of chips, it will set the `max-width: 0px` and transition that property over 200ms. Should I add a comment on what this value is?